rad/prelude/list.rad

```diff
@@ -74,3 +75,18 @@
   [ (filter (fn [x] (< x 10)) (list 3 10 11))
      ==> (3) ]
 )
+
+;; take-while
+
+(def-rec take-while
+  "Returns all elements of a list `ls` until one does not satisfy `pred`"
```

Comment:
After #277 this should work on vectors as well. Might be worth waiting on it, and then changing the docs here to say `sequence`, and the tests to include vectors.